File: truevfs-ext/truevfs-ext-pacemaker/src/main/scala/net/java/truevfs/ext/pacemaker/PaceManagerView.scala
/*
 * Copyright (C) 2005-2015 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.pacemaker

import javax.annotation.concurrent.ThreadSafe
import javax.management.MBeanAttributeInfo
import javax.management.MBeanInfo
import net.java.truevfs.comp.jmx.JmxManagerView

/**
 * A view for a [[PaceManager]].
 *
 * @author Christian Schlichtherle
 */
@ThreadSafe
private final class PaceManagerView(manager: PaceManager)
extends JmxManagerView[PaceManager](classOf[PaceManagerMXBean], manager)
with PaceManagerMXBean {

  protected override def getDescription(info: MBeanInfo) = "A pace maker for the file system manager."

  protected override def getDescription(info: MBeanAttributeInfo) = {
    info.getName match {
      case "MaximumFileSystemsMounted" => "The maximum number of mounted file systems."
      case _                           => super.getDescription(info)
    }
  }

  override def getMaximumFileSystemsMounted = manager.max
  override def setMaximumFileSystemsMounted(max: Int) { manager.max = max }
}

File: truevfs-ext/truevfs-ext-pacemaker/src/main/java/net/java/truevfs/ext/pacemaker/PaceManagerView.java
/*
 * Copyright (C) 2005-2020 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.pacemaker;

import net.java.truevfs.comp.jmx.JmxManagerView;

import javax.management.MBeanAttributeInfo;
import javax.management.MBeanInfo;

/**
 * A view for a {@link PaceManager}.
 * This class is thread-safe.
 *
 * @author Christian Schlichtherle
 */
final class PaceManagerView extends JmxManagerView<PaceManager> implements PaceManagerMXBean {

    public PaceManagerView(PaceManager manager) {
        super(manager, PaceManagerMXBean.class);
    }

    @Override
    protected String getDescription(MBeanInfo info) {
        return "A pace maker for the file system manager.";
    }

    @Override
    protected String getDescription(MBeanAttributeInfo info) {
        return "MaximumFileSystemsMounted".equals(info.getName())
                ? "The maximum number of mounted file systems."
                : super.getDescription(info);
    }

    @Override
    public int getMaximumFileSystemsMounted() {
        return manager.getMaximumSize();
    }

    @Override
    public void setMaximumFileSystemsMounted(int maxMounted) {
        manager.setMaximumSize(maxMounted);
    }
}
